okhttp/src/jvmTest/java/okhttp3/CacheControlJvmTest.kt

```diff
@@ -0,0 +1,208 @@
+        Assertions.assertThat(cacheControl.onlyIfCached).isFalse
+        Assertions.assertThat(cacheControl.mustRevalidate).isFalse
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun parse() {
+        val header = ("no-cache, no-store, max-age=1, s-maxage=2, private, public, must-revalidate, "
+                + "max-stale=3, min-fresh=4, only-if-cached, no-transform")
+        val cacheControl = parse(
+            Headers.Builder()
+                .set("Cache-Control", header)
+                .build()
+        )
+        Assertions.assertThat(cacheControl.noCache).isTrue
+        Assertions.assertThat(cacheControl.noStore).isTrue
+        Assertions.assertThat(cacheControl.maxAgeSeconds).isEqualTo(1)
+        Assertions.assertThat(cacheControl.sMaxAgeSeconds).isEqualTo(2)
+        Assertions.assertThat(cacheControl.isPrivate).isTrue
+        Assertions.assertThat(cacheControl.isPublic).isTrue
+        Assertions.assertThat(cacheControl.mustRevalidate).isTrue
+        Assertions.assertThat(cacheControl.maxStaleSeconds).isEqualTo(3)
+        Assertions.assertThat(cacheControl.minFreshSeconds).isEqualTo(4)
+        Assertions.assertThat(cacheControl.onlyIfCached).isTrue
+        Assertions.assertThat(cacheControl.noTransform).isTrue
+        Assertions.assertThat(cacheControl.toString()).isEqualTo(header)
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun parseIgnoreCacheControlExtensions() {
+        // Example from http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.9.6
+        val header = "private, community=\"UCI\""
+        val cacheControl = parse(
+            Headers.Builder()
+                .set("Cache-Control", header)
+                .build()
+        )
+        Assertions.assertThat(cacheControl.noCache).isFalse
+        Assertions.assertThat(cacheControl.noStore).isFalse
+        Assertions.assertThat(cacheControl.maxAgeSeconds).isEqualTo(-1)
+        Assertions.assertThat(cacheControl.sMaxAgeSeconds).isEqualTo(-1)
+        Assertions.assertThat(cacheControl.isPrivate).isTrue
+        Assertions.assertThat(cacheControl.isPublic).isFalse
+        Assertions.assertThat(cacheControl.mustRevalidate).isFalse
+        Assertions.assertThat(cacheControl.maxStaleSeconds).isEqualTo(-1)
+        Assertions.assertThat(cacheControl.minFreshSeconds).isEqualTo(-1)
+        Assertions.assertThat(cacheControl.onlyIfCached).isFalse
+        Assertions.assertThat(cacheControl.noTransform).isFalse
+        Assertions.assertThat(cacheControl.immutable).isFalse
+        Assertions.assertThat(cacheControl.toString()).isEqualTo(header)
+    }
+
+    @Test
+    fun parseCacheControlAndPragmaAreCombined() {
+        val headers = headersOf("Cache-Control", "max-age=12", "Pragma", "must-revalidate", "Pragma", "public")
+        val cacheControl = parse(headers)
+        Assertions.assertThat(cacheControl.toString()).isEqualTo("max-age=12, public, must-revalidate")
+    }
+
+    // Testing instance equality.
```

Comment:
nit: equality is kind of the wrong word here, we’re testing instance identity?

Reply:
This comment made more sense when it was on an `@SuppressWarnings` thing. Right now the comment is pretty unnecessary; the `isSameAs` does it all.

Reply:
ALSO the redundant string construction that motivated the warning is missing? I think that’s intended to confirm we get the same instance, working against `String.intern()` on the JVM?

Reply:
Seems useful, particularly if these are from header compression at http layer.  I'll remove comment but keep test.